core/src/main/java/hudson/Functions.java

```diff
@@ -1912,14 +1912,12 @@ public static String joinPath(String... components) {
      * @return the absolute URL for accessing the build console for the executable, or null if there is no build associated with the executable
      * @since 2.433
      */
-    public static @CheckForNull String getConsoleUrl(Queue.Executable executable) {
-        if (executable == null) {
-            return null;
-        } else if (executable instanceof Run) {
-            return ConsoleUrlProvider.getRedirectUrl((Run<?, ?>) executable);
+    public static @CheckForNull String getConsoleUrl(Object executable) {
```

Comment:
Can this be adjusted to accept the new interface instead? Maybe also worth adjusting the Javadoc to replace mentions of `executable` with something more generic.

Reply:
done